app/services/data_migrations/backfill_international_degrees_subjects_uuid.rb

```diff
@@ -0,0 +1,41 @@
+module DataMigrations
+  class BackfillInternationalDegreesSubjectsUuid
+    TIMESTAMP = 20220519142519
+    MANUAL_RUN = false
+    AFTER_NEW_DEGREE_FLOW_RELEASE = '2022-05-04T13:51:00Z'.freeze
+
+    def change
+      records.find_each do |qualification|
+        subject = subject_for(qualification)
+
+        if subject.present?
+          qualification.update_columns(
+            degree_subject_uuid: subject.id,
+          )
+        end
+      end
+    end
+
+    def dry_run
+      total = records.select do |record|
+        subject_for(record).present?
+      end
+
+      # rubocop:disable Rails/Output
+      puts "There are #{total.size} international records with subject in the reference data"
+      # rubocop:enable Rails/Output
+    end
+
+  private
+
+    def records
+      ApplicationQualification.degree
+      .where.not(institution_country: nil)
+      .where('created_at > ?', AFTER_NEW_DEGREE_FLOW_RELEASE)
```

Comment:
Since updating qualifications touches application choices, is there a risk of touching applications from a previous cycle?

Reply:
The new degree low release was less than a month ago so there are hundreds of international degrees from the last month only to be updated and 60% find the UUIDs from the reference data. We used `update_columns` to not trigger any callback or unnecessary code. Concluding, no risk :)